### PR TITLE
Increased test Coverage of pkg/chrootarchive

### PR DESCRIPTION
**Pull request by raja-sami-10p**
Signed-off-by: Raja Sami <raja.sami@tenpearls.com>

**- What I did**
Increased unit test coverage of pkg/chrootarchive

**- How to verify it**
```sh
make test-unit
```

**- A picture of a cute animal (not mandatory but encouraged)**
:bird:

### CODE REVIEW

**Comment by vieux** [comment]
LGTM

File: pkg/chrootarchive/archive_test.go [hunk 1]
```diff
@@ -39,6 +41,80 @@ func UntarPath(src, dst string) error {
...
+
+	destination := filepath.Join(tempDir, "src")
+
+	err = system.MkdirAll(destination, 0700)
+	require.NoError(t, err)
+
+	err = ioutil.WriteFile(filepath.Join(destination, "test"), []byte("test un compressed"), 0644)
+	require.NoError(t, err)
+
+	err = UntarUncompressed(nil, destination, &archive.TarOptions{ExcludePatterns: []string{"testuncompresss"}})
```
Comment (thaJeztah) [inline pkg/chrootarchive/archive_test.go]: Although it doesn't really _hurt_, all the exclusion patterns seem to be covered already by other tests (the only different between `UntarUncompressed` and `Untar` is this bit https://github.com/moby/moby/blob/93fbdb69acf9248283a91a1c5c6ea24711c26eda/pkg/chrootarchive/archive.go#L60-L67, of which only the error-case seems to be not covered.

File: pkg/chrootarchive/archive_test.go [hunk 2]
```diff
@@ -39,6 +41,80 @@ func UntarPath(src, dst string) error {
...
+	assert.EqualError(t, err, "Empty archive")
+
+	stream, err := archive.Tar(destination, archive.Uncompressed)
+	require.NoError(t, err)
+
+	err = UntarUncompressed(stream, destination, &archive.TarOptions{ExcludePatterns: []string{"testuncompresss"}})
+	assert.NoError(t, err)
+
+	stream, err = archive.Tar(destination, archive.Uncompressed)
```
Comment (thaJeztah) [inline pkg/chrootarchive/archive_test.go]: What is being tested here? `archive.Tar()`, `archive.DecompressStream()` are all part of [`pkg/archive`](https://github.com/moby/moby/blob/93fbdb69acf9248283a91a1c5c6ea24711c26eda/pkg/archive/archive.go)?

File: pkg/chrootarchive/archive_test.go [hunk 3]
```diff
@@ -39,6 +41,80 @@ func UntarPath(src, dst string) error {
...
+	stream, err := archive.Tar(destination, archive.Uncompressed)
+	require.NoError(t, err)
+
+	_, err = ApplyUncompressedLayer(destination, stream, &archive.TarOptions{ExcludePatterns: nil})
+	assert.NoError(t, err)
+
+	_, err = ApplyUncompressedLayer(destination, stream, &archive.TarOptions{ExcludePatterns: []string{"testUnCompressed"}})
+	assert.NoError(t, err)
+
+	stream, err = archive.Tar(destination, archive.Uncompressed)
```
Comment (thaJeztah) [inline pkg/chrootarchive/archive_test.go]: Same here; I don't see what's being tested here. Actually, this looks like an exact copy of what's in `TestUnTarUncompressed` above

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Left some questions

**Comment by cpuguy83** [comment]
I don't think this test is valid, and surprised that it didn't break the rest of the test suite.
If we want to test this function it needs to be done via re-exec, and will be pretty hard to test.

**Comment by raja-sami-10p** [comment]
@cpuguy83, can you please help me understand how this test should have broken the rest of the test suites?

**Comment by cpuguy83** [comment]
@raja-sami-10p Because the function being tested is creating a new mount namespace, and then doing a pivot_root to an empty directory.

**Comment by aaronlehmann** [comment]
Since it sounds like this test is invalid, should we close the PR?

**Comment by cpuguy83** [comment]
Yes, let's close this one.

Thanks all!

**Pull request closed without merging**